clang/lib/Headers/hlsl/hlsl_intrinsics.h

```diff
@@ -174,6 +174,31 @@ const inline float distance(__detail::HLSL_FIXED_VECTOR<float, N> X,
                             __detail::HLSL_FIXED_VECTOR<float, N> Y) {
   return __detail::distance_vec_impl(X, Y);
 }
+//===----------------------------------------------------------------------===//
+// dst builtins
+//===----------------------------------------------------------------------===//
+
+/// \fn fvector dst( fvector, fvector)
+/// \brief Returns the length of a vector
+/// \param src0 [in] The first vector contain {_, d*d, d*d, _}
+/// \param src1 [in] The second vector contain {_, 1/d, _, 1/d}
+///
+/// Return the computed distance vector contain {1, d, d*d, 1/d}
+
+_HLSL_16BIT_AVAILABILITY(shadermodel, 6.2)
+const inline vector<half, 4> dst(vector<half, 4> src0, vector<half, 4> src1) {
+  return __detail::dst_impl(src0, src1);
+}
+
+const inline vector<float, 4> dst(vector<float, 4> src0,
```

Comment:
The scalar argument handling looks correct. Would like others to agree with me. DXC code gen is on the left. yours is on the right
Your code: https://hlsl.godbolt.org/z/576aP5eq4
DXC: https://hlsl.godbolt.org/z/hhP7o368T
The first and third functions look correct. The second function you will notice your code has an extra extract and insert that dxc does not have. I think this is fine because it looks to be what  shufflevector would have done.
![image](https://github.com/user-attachments/assets/267ff718-64e1-4b25-b653-6acc0ae9f3f7)